resources/js/features/home/components/+sidebar/TopLinks/TopLinks.tsx

```diff
@@ -11,54 +11,100 @@ import {
 import { PiMedalFill } from 'react-icons/pi';
 import { SiDiscord, SiPatreon } from 'react-icons/si';
 
-import { BaseButton, baseButtonVariants } from '@/common/components/+vendor/BaseButton';
-
-// TODO if config.services.discord.invite_id
-// TODO if config.services.patreon.user_id
-// TODO tracking
+import { baseButtonVariants } from '@/common/components/+vendor/BaseButton';
+import { usePageProps } from '@/common/hooks/usePageProps';
+import { buildTrackingClassNames } from '@/common/utils/buildTrackingClassNames';
+import { cn } from '@/utils/cn';
 
 export const TopLinks: FC = () => {
+  const { config } = usePageProps<App.Http.Data.HomePageProps>();
+
   const { t } = useLaravelReactI18n();
 
   return (
     <div className="flex flex-col gap-2.5">
-      <a href={route('download.index')} className={baseButtonVariants({ size: 'sm' })}>
+      <a
+        href={route('download.index')}
+        className={cn(
+          baseButtonVariants({ size: 'sm' }),
+          buildTrackingClassNames('Click Top Link Download Emulator'),
+        )}
+      >
         <LuDownload className="mr-2 h-4 w-4 text-sky-400" />
         {t('Download Emulator')}
       </a>
 
       <div className="flex flex-col gap-2.5 sm:grid sm:grid-cols-2 lg:flex">
-        <a href="/globalRanking.php" className={baseButtonVariants({ size: 'sm' })}>
+        <a
+          href="/globalRanking.php"
+          className={cn(baseButtonVariants({ size: 'sm' }), 'Click Top Link Global Points Ranking')}
+        >
           <PiMedalFill className="mr-2 h-4 w-4 text-amber-400" />
           {t('Global Points Ranking')}
         </a>
 
-        <a href={route('ranking.beaten-games')} className={baseButtonVariants({ size: 'sm' })}>
+        <a
+          href={route('ranking.beaten-games')}
+          className={cn(
+            baseButtonVariants({ size: 'sm' }),
+            'Click Top Link Global Beaten Games Ranking',
+          )}
```

Comment:
I'm getting a 500 error trying to access this page (happens on master too), but not on stage, so must be something in my environment. It's not writing an error log, so I don't know how to diagnose.

Reply:
On master, does it also happen on the normal home page, or just /demo/home ?

Reply:
If you try manually running `composer install` does that change anything?

Reply:
It's not the link, it's the page itself.
![image](https://github.com/user-attachments/assets/f87befcd-44e9-44d1-956f-5e3a77526f2b)

`composer install` did not change the behavior.

Reply:
Ah, I misinterpreted entirely earlier and understand the issue very clearly now.

It's an out-of-memory error and unrelated to the changes in this PR. I bet if you check your terminal, you'll see it yelling about running out of memory / memory allocation / etc when you visit the unfiltered beaten games leaderboard. If you add a filter manually in the address bar, the page will magically start working again.

When we built the beaten games leaderboard, the memory allocation of the page was fine. Now that we've grown, the total unfiltered list uses more memory. I'll need to hunt that down.

Other pages exhibit this problem too, for example: http://localhost:64000/codenotes.php?g=2721
This is the code notes page for Dragon Quest VIII. It has an ungodly high amount of code notes - enough to crash our local envs without bumping the memory limit.

You can get around this by bumping the PHP memory limit. To do so, open _docker/8.2/php.ini_ and change `memory_limit` to something like 256M. For the memory limit change to take effect, you have to rebuild your Sail images like so:
```bash
sail down
sail build --no-cache
sail up
```